provider/cloudflare/cloudflare.go

```diff
@@ -455,14 +449,19 @@ func (p *CloudFlareProvider) submitChanges(ctx context.Context, changes []*cloud
 					failedChange = true
 					log.WithFields(logFields).Errorf("failed to delete record: %v", err)
 				}
-				chID, oldCh := p.getCustomHostnameIDbyOrigin(chs, change.ResourceRecord.Name)
+				if change.CustomHostname.Hostname == "" {
+					continue
+				}
+				log.WithFields(logFields).Infof("Deleting custom hostname %v", change.CustomHostname.Hostname)
+				chID, _ := p.getCustomHostnameOrigin(chs, change.CustomHostname.Hostname)
 				if chID == "" {
+					log.WithFields(logFields).Infof("Custom hostname %v not found", change.CustomHostname.Hostname)
```

Comment:
not covered, could we have a test with capturing logs? https://github.com/kubernetes-sigs/external-dns/blob/4b2fd495d944f9a67cdeb5260db95c5f2367d8bf/internal/testutils/log.go#L36

![Screenshot 2025-03-07 at 08 46 11](https://github.com/user-attachments/assets/2883e657-c628-441c-8eac-23f4ae190839)

Reply:
This is largely innocuous case when the custom hostname has changed after the plan calculation.
The message is informational and is not expected to really happen, because the custom hostname is immutable in Cloudflare.
Appreciating the value of the tests coverage, however, I've added a separate test with the `LogsToBuffer` helper.
It's relatively messy, and has to intentionally corrupt the internal custom hostnames structure to have the condition to trigger.